text/0000-innerhtml-binding.md

```diff
@@ -0,0 +1,148 @@
+---
+title: InnerHTML Bindings template binding
+status: DRAFTED
+created_at: 2019-10-03
+updated_at: 2021-03-08
+pr: https://github.com/salesforce/lwc/pull/2256
+champion: Philippe Riand, Pierre-Marie Dartus
+---
+
+# InnerHTML Bindings for SSR
+
+## Summary
+
+Applications dealing with rich content, like commerce apps, need to render this content as raw HTML, coming from a database or a CMS. Today, there is no way in LWC to rendered raw HTML content both on the browser and the server. This proposal introduces the `lwc:inner-html` template directive to inject HTML content.
+
+### Basic example
+
+The following template renders raw HTML content inside the `<div>` HTML element:
+
+**example.js**
+```js
+import { LightningElement } from 'lwc';
+
+export default class Example extends HTMLElement {
+  content = 'Hello <strong>world</strong>!';
+}
+```
+
+**example.html**
+```html
+<template>
+  <div lwc:inner-html={content} />
+</template>
+```
+
+**Output:**
+
+```html
+<x-example>
+  # shadow-root
+  |   <div>
+  |     Hello <strong>world</strong>
+  |   </div>
+</x-example>
+```
+
+## Motivation
+
+It is common in customer-facing apps to render rich content coming from a CMS. As of today, this capability is supported by acting directly with the DOM within `renderedCallback()`. The templating syntax does not provide any binding capability that renders some content as raw HTML.
+
+The reliance on the `renderedCallback()` life-cycle hook to render raw HTML content is a real issue for server-side rendering. In SSR, the `renderedCallback()` is never invoked. There is currently no straightforward way to inject raw HTML content for server-side rendered components.
+
+This proposal introduces a new `lwc:inner-html` directive to inject raw HTML content that would work both in the browser and on the server.
+
+## Detailed design
+
+### Runtime behavior
+
+At runtime the `lwc:inner-html` directive binds the directive value to the `Element.innerHTML` property. During each rendering cycle, the LWC engine diff the current value with its previous value. If the value changes between two rendering cycles, the LWC engine sets the element `innerHTML` property to the new value.

```

Comment:
```suggestion
At runtime the `lwc:inner-html` directive binds the directive value to the `Element.innerHTML` property. During each rendering cycle, the LWC engine diff the current string value with its previous value. If the value changes between two rendering cycles, the LWC engine sets the element `innerHTML` property to the new value.

```

This might be confusing as you can diff the generated DOM tree.